src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/EventService.java

```diff
@@ -67,43 +82,275 @@ static EventDTO createEventDTOFromEvent(Event event) {
     return dto;
   }
 
-  Event updateEvent(UUID collexUuid, String tag, Date date) throws CTPException;
+  @Autowired private CollectionExerciseService collectionExerciseService;
+
+  @Autowired private EventRepository eventRepository;
+
+  @Autowired private EventChangeHandler[] changeHandlers = {};
+
+  @Autowired private List<EventValidator> eventValidators;
+
+  @Autowired private Scheduler scheduler;
 
-  Event getEvent(UUID collexUuid, String tag) throws CTPException;
+  @Autowired private List<ActionRuleCreator> actionRuleCreators;
 
-  Event getEvent(UUID eventId) throws CTPException;
+  @Autowired private List<ActionRuleUpdater> actionRuleUpdaters;
 
-  Event deleteEvent(UUID collexUuid, String tag) throws CTPException;
+  public Event createEvent(EventDTO eventDto) throws CTPException {
+    UUID collexId = eventDto.getCollectionExerciseId();
+    CollectionExercise collex = getCollectionExercise(collexId, Fault.RESOURCE_NOT_FOUND);
+    Event existing =
+        this.eventRepository.findOneByCollectionExerciseAndTag(collex, eventDto.getTag());
 
-  List<Event> getOutstandingEvents();
+    if (existing != null) {
+      throw new CTPException(
+          Fault.RESOURCE_VERSION_CONFLICT,
+          String.format(
+              "Event %s already exists for collection exercise %s",
+              eventDto.getTag(), collex.getId()));
+    }
+
+    Event event = new Event();
+
+    event.setCollectionExercise(collex);
+    event.setTag(eventDto.getTag());
+    event.setId(UUID.randomUUID());
+    event.setTimestamp(new Timestamp(eventDto.getTimestamp().getTime()));
+    event.setCreated(new Timestamp(new Date().getTime()));
```

Comment:
I think there's a helper function to get a Timestamp for now

Reply:
This isn't functionality that was added this PR so will leave this one for another time